CompanyApi/Controllers/CompaniesController.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Metadata.Ecma335;
+using System.Threading.Tasks;
+using CompanyApi.Models;
+using Microsoft.AspNetCore.CookiePolicy;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace CompanyApi.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class CompaniesController : ControllerBase
+    {
+        private static SortedDictionary<string, Company> companies = new SortedDictionary<string, Company>();
+        private static string companyID = "0";
+
+        [HttpDelete("clear")]
+        public async Task ClearCompany()
+        {
+            companyID = "0";
+            companies.Clear();
+        }
+
+        [HttpPost]
+        public async Task<ActionResult<Company>> AddCompany(Company company)
+        {
+            if (companies.Count != 0 && companies.Where(idCompanyPair => idCompanyPair.Value.Name == company.Name).ToList().Count > 0)
+            {
+                return Conflict();
+            }
+
+            string companyID = GenerateCompanyID();
+            Company newCompany = new Company(companyID, company.Name);
+            newCompany.Employees = company.Employees;
+            companies[companyID] = newCompany;
+            return CreatedAtRoute(nameof(GetCompanyByID), new { id = newCompany.CompanyID }, newCompany);
+        }
+
+        [HttpGet]
+        public List<Company> GetAllCompany()
+        {
+            return companies.Select(idCompanyPair => idCompanyPair.Value).ToList();
+        }
+
+        [HttpGet("{id}", Name = "GetCompanyByID")]
+        public Company GetCompanyByID(string id)
```

Comment:
there are two id in requirement, is it better to specify which id you mean? the same questions below